cassandraprotocol/frame/encoder.go

```diff
@@ -10,24 +10,25 @@ import (
 	"io"
 )
 
-const encodedHeaderLength = 9
-
-func (c *codec) Encode(frame *Frame) (*bytes.Buffer, error) {
+func (c *Codec) Encode(frame *Frame, dest io.Writer) error {
```

Comment:
I was about to ask you about this, I noticed that we weren't using `io.Writer` but  `bytes.Buffer` which made it a bit harder to use but you already changed it 😄 nice

Reply:
Absolutely, using `io.Writer` and `io.Reader` makes it extremely simple to read and write to `net.Conn` since it satisfies both interfaces. And as a bonus, this means that we don't need to allocate any temporary buffer.